redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3415,931 @@ inline void requestRoutesPostCodesEntry(App& app)
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] redfishUri - The iterator to fill redfish uri with its "Id".
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else nothing in redfish response.
+ *
+ * @note - This function will make sure to assign the redfish uri with Ids
+ *         if it contains "<str>".
+ *       - This function will run recursively until finding the redfish uri
+ *         from the list by using the given dbus object path.
+ */
+inline void
+    getRedfishUriFromList(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const size_t entryJsonIdx,
+                          const sdbusplus::message::object_path& dbusObjPath,
+                          RedfishUriListType::const_iterator redfishUriIt)
+{
+    if (redfishUriIt == redfishUriList.end())
+    {
+        BMCWEB_LOG_ERROR
+            << "Failed to get Redfish URI for the given DBus object path: "
+            << dbusObjPath.str;
+        return;
+    }
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, entryJsonIdx, dbusObjPath,
+         redfishUriIt](const boost::system::error_code ec,
+                       const GetObjectType& objType) mutable {
+            if (ec || objType.empty())
+            {
+                // Look the next element from redfish uri list
+                getRedfishUriFromList(asyncResp, entryJsonIdx, dbusObjPath,
+                                      std::next(redfishUriIt));
+            }
+            else
+            {
+                std::string redfishUri = std::get<1>(*redfishUriIt) + "/" +
+                                         getIsolatedHwItemId(dbusObjPath);
+                if (redfishUri.find("<str>") == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    /**
+                     * For example,
+                     * "/redfish/v1/Systems/system/Processors/<str>/ \
+                     * SubProcessors/core0"
+                     */
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
+                                          redfishUri);
+                }
+            }
+        },
+        objMapperSerName, objMapperObjPath, objMapperIface, "GetObject",
+        dbusObjPath.str, std::get<0>(*redfishUriIt));
```

Comment:
common error 5

Reply:
Agree on the common error 5 but, here I am using for tuple, not for the vector. 
I think tuple won't throw an exception. FYR https://en.cppreference.com/w/cpp/utility/tuple/get - Even that bmcweb common error did not mention for the `tuple` - I am missing anything on this?